integrations/pinecone/src/haystack_integrations/document_stores/pinecone/document_store.py

```diff
@@ -295,6 +296,29 @@ def _convert_query_result_to_documents(self, query_result: Dict[str, Any]) -> Li
 
         return documents
 
+    @staticmethod
+    def check_metadata(document: Document):
+        def valid_type(value: Any):
+            return isinstance(value, METADATA_SUPPORTED_TYPES) or (
+                isinstance(value, list) and all(isinstance(i, str) for i in value)
+            )
+
+        if document.meta:
+            discarded_keys = []
+            for key, value in document.meta.items():
+                if not valid_type(value):
+                    discarded_keys.append(key)
+                    document.meta[key] = "IGNORED"
```

Comment:
I would prefer to discard the meta field.
Consider also that Pinecone is a paid service, so we waste (some) resources.